control_plane/attachment_service/src/service.rs

```diff
@@ -1221,14 +1221,19 @@ impl Service {
         Ok(())
     }
 
-    /// This API is used by the cloud control plane to do coarse-grained control of tenants:
+    /// This API is used by the cloud control plane to migrate unsharded tenants that it created
+    /// directly with pageservers into this service.
+    ///
+    /// Cloud control plane MUST NOT continue issuing GENERATION NUMBERS for this tenant once it
+    /// has attempted to call this API. Failure to oblige to this rule may lead to S3 corruption.
+    /// Think of the first attempt to call this API as a transfer of absolute authority over the
+    /// tenant's source of generation numbers.
+    ///
+    /// The mode in this request coarse-grained control of tenants:
```

Comment:
In b20f33bd1dd2a0bfa1ebd85ec3c5ec183cd9d575 I've made this stricter about generation numbers, so that we only pay attention to them in attached modes.

The behavior of callers isn't something we can fully 
 enforce here, because they might attach a tenant elsewhere without our knowledge. 
 https://github.com/neondatabase/cloud/issues/10417 tracks making this stricter in the cloud control plane.